test/logging_test.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "gtest/gtest.h"
+#include "include/proxy-wasm/wasm.h"
+#include "test/utility.h"
+
+namespace proxy_wasm {
+
+INSTANTIATE_TEST_SUITE_P(WasmEngines, TestVm, testing::ValuesIn(getWasmEngines()),
+                         [](const testing::TestParamInfo<std::string> &info) {
+                           return info.param;
+                         });
+
+// TestVm is parameterized for each engine and creates a VM on construction.
+TEST_P(TestVm, HttpLogging) {
+  // Read the wasm source.
+  auto source = readTestWasmFile("http_logging.wasm");
+  ASSERT_FALSE(source.empty());
+
+  // Create a WasmBase and load the plugin.
+  auto wasm = std::make_shared<TestWasm>(std::move(vm_));
```

Comment:
Question to @PiotrSikora: do you prefer that tests/code generally use `createWasm()` to instantiate the Wasm object, or is direct initialization like this fine? Initialization via calls to individual WasmBase methods like this ends up being shorter, since no need to create factories.

Reply:
FWIW there's precedent for direct initialization in the other unit tests, and here I think it helps to illustrate the minimal use case for invoking ProxyWasm.

Reply:
Resolving for now in case the comment is blocking approval

Reply:
> Question to @PiotrSikora: do you prefer that tests/code generally use `createWasm()` to instantiate the Wasm object, or is direct initialization like this fine? Initialization via calls to individual WasmBase methods like this ends up being shorter, since no need to create factories.

This is fine (see: #270), but I don't think we need `std::make_shared()` here.

> Resolving for now in case the comment is blocking approval

That's kind of the point of this feature (also, it's blocking merge, not approval).

Reply:
> I don't think we need std::make_shared() here.

All the relevant constructors take shared_ptrs:
- WasmBase -> WasmHandleBase
- WasmHandleBase + PluginBase -> PluginHandleBase
- PluginHandleBase -> ContextBase

I removed the unique_ptr around TestContext below.

Reply:
Sorry, I'm unclear on the etiquette. Can I resolve this now that you've approved?